service/controller/v1/resource/drainer/create.go

```diff
@@ -182,7 +182,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 		drainerConfig.Status.Conditions = append(drainerConfig.Status.Conditions, drainerConfig.Status.NewDrainedCondition())
 
-		_, err := r.g8sClient.CoreV1alpha1().DrainerConfigs(drainerConfig.GetNamespace()).Update(&drainerConfig)
+		_, err := r.g8sClient.CoreV1alpha1().DrainerConfigs(drainerConfig.GetNamespace()).UpdateStatus(&drainerConfig)
```

Comment:
For my interest: how was it working before?

Reply:
I can imagine that there is some client weirdness that made it work somehow. I have really no idea.

Reply:
Because of this: https://github.com/giantswarm/node-operator/blob/master/helm/node-operator-chart/templates/rbac.yaml#L35

I don't think there's anything special with CRD Statuses except ability define stricter RBAC rules. One can define separate rules for `<crd>/status` so write/update/patch operations could be removed from operator reconciling on a specific CRD and then only added to status part of that.

I'd be actually interested in working on that so gonna create an issue on that.

Reply:
When the CRD defines the status subresource you can only patch the status via `UpdateStatus` and calls to `Update` do not affect it. When the CRD does not define the status subresource you can apparently still update the status via `Update` and a `UpdateStatus` method does not exist.